### PR TITLE
Fixed block response limit check

### PR DESCRIPTION
**Pull request by arkpar**
When computing block response size, the code was incorrectly counting extrinsics rather than bytes.

Also decreased `MAX_BLOCKS_TO_REQUEST` so that the nodes can sync without waiting for others to upgrade and tweaked timeout value.

Fixes https://github.com/paritytech/polkadot/issues/3778

### CODE REVIEW

File: client/network/src/block_request_handler.rs [hunk 1]
```diff
@@ -355,7 +355,8 @@ impl<B: BlockT> BlockRequestHandler<B> {
 				indexed_body,
 			};
 
-			total_size += block_data.body.len();
+			total_size += block_data.body.iter().map(|ex| ex.len()).sum::<usize>();
+			total_size += block_data.indexed_body.iter().map(|ex| ex.len()).sum::<usize>();
```
Comment (gilescope) [inline client/network/src/block_request_handler.rs:359]: Might this work well as a size() function on block_data?

File: client/network/src/block_request_handler.rs [hunk 2]
```diff
@@ -62,7 +62,7 @@ pub fn generate_protocol_config(protocol_id: &ProtocolId) -> ProtocolConfig {
 		name: generate_protocol_name(protocol_id).into(),
 		max_request_size: 1024 * 1024,
 		max_response_size: 16 * 1024 * 1024,
-		request_timeout: Duration::from_secs(40),
+		request_timeout: Duration::from_secs(20),
```
Comment (bkchr) [inline client/network/src/block_request_handler.rs:65]: Why did you change the timeout?

**Review by bkchr — APPROVED** [review]
Adding a test would be nice

File: client/network/src/block_request_handler.rs (hunk 2 above)
Comment (tomaka) [reply client/network/src/block_request_handler.rs:65]: It's a bit off-topic for this PR, but this timeout is too high anyway.

**Review by tomaka — APPROVED** [review]

File: client/network/src/block_request_handler.rs (hunk 2 above)
Comment (arkpar) [reply client/network/src/block_request_handler.rs:65]: Right it was too high in the first place, and since I've also decreased the number of requested blocks the timeout should be readjusted.

File: client/network/src/block_request_handler.rs (hunk 1 above)
Comment (arkpar) [reply client/network/src/block_request_handler.rs:359]: `BlockData` struct is auto generated from protobuf description. Also, we don't need all of the field to be sized here, just the bodies. The rest do not vary much and should definitely fit in the margin.

**Comment by arkpar** [comment]
Added a test and fixed response start block detection. Rather than trusting that it matches the request it should be detected from the response.

**Comment by bkchr** [comment]
@arkpar did you missed to push the test changes?

**Comment by bkchr** [comment]
bot merge
